app/models/concerns/availability_status_concern.rb

```diff
@@ -0,0 +1,25 @@
+module AvailabilityStatusConcern
+  extend ActiveSupport::Concern
+
+  included do
+    before_update :update_status
+  end
+
+  private
+
+  STATUS_RELATED_ATTRS = [
+    "availability_status",
+    "last_checked_at",
+    "updated_at",
+    "last_available_at",
+    "availability_status_error"
+  ].freeze
+
+  def update_status
+    updated_attributes = self.changed - STATUS_RELATED_ATTRS
+    unless updated_attributes.empty?
+      self.availability_status = nil
+      self.last_checked_at = nil
```

Comment:
availability status error can be also cleared

Reply:
That’s a good idea. Requested code has been imported. Thanks